test/Chainweb/Test/Pact/ModuleCacheOnRestart.hs

```diff
@@ -112,16 +113,17 @@ testCoinbase iobdb = (initPayloadState >> doCoinbase,snapshotCache)
       void $ execValidateBlock mempty nextH (payloadWithOutputsToPayloadData pwo)
 
 -- | Interfaces can't be upgraded, but modules can, so verify hash in that case.
-justModuleHashes :: ModuleCache -> HM.HashMap ModuleName (Maybe ModuleHash)
-justModuleHashes = HM.map $ \v -> preview (_1 . mdModule . _MDModule . mHash) v
+justModuleHashes :: ModuleInitCache -> HM.HashMap ModuleName (Maybe ModuleHash)
+justModuleHashes = upd. snd . last . M.toList where
```

Comment:
I think, future versions of GHC will require white space around function composition `.` (not sure if on both sides, though).

```suggestion
justModuleHashes = upd . snd . last . M.toList where
```